article.md

```diff
@@ -108,19 +163,17 @@ if __name__ == "__main__":
     asyncio.get_event_loop().run_until_complete(fetch_page('http://zestedesavoir.com/', 'out.html'))
 ```
 
-Mais les modifications sont plus importantes que ces simples synonymes. Tout d'abord une coroutine déclaré avec `async` n'est PAS un générateur. Même si en interne les deux types de fonctions partagent une grande partie de leur implémentation, il s'agit de types différents et il est possible que les différences se creusent dans les prochaines versions de Python. Pour marquer cette différence et le fait que des générateurs sont une forme retreinte de coroutines, il est possible dans Python 3.5 d'utiliser des générateurs partout où une coroutine est attendu, mais pas l'inverse. Le module *asyncio* continue ainsi à supporter les deux formes.
+Mais les modifications sont plus importantes que ces simples synonymes. Tout d'abord une coroutine déclaré avec `async` n'est PAS un générateur. Même si en interne les deux types de fonctions partagent une grande partie de leur implémentation, il s'agit de constructions du langage différentes et il est possible que les différences se creusent dans les prochaines versions de Python. Pour marquer cette différence et le fait que des générateurs sont une forme retreinte de coroutines, il est possible dans Python 3.5 d'utiliser des générateurs partout où une coroutine est attendu, mais pas l'inverse. Le module *asyncio* continue ainsi à supporter les deux formes.
```

Comment:
les coroutines de python 3.5 ou les generateurs décorés sont de l'exterieur identique et s'utilisent de façon identique et transparentes. C'est la meme chose. Ce sont deux façon d'écrires des coroutines. les générateur avec decorateur ont l'avantage d'etre dispo dès python 3.4, les `async def` elles ont l'avantage de permettre les `async for` et `async with`.

Mais ce sont toutes les deux la meme chose, seul deux façon de les écrire.